pkg/server/server.go

```diff
@@ -47,6 +52,13 @@ func New(configuration options.Configuration) (controllerruntime.Manager, error)
 		return nil, fmt.Errorf("can't create client for api %s: %w", v1alpha1.SchemeGroupVersion, err)
 	}
 
+	// Check if metacontroller can successfully communicate with the K8s API server
+	// If metacontroller is in a service mesh, this serves as a check that the sidecar is healthy
+	err = k8sCommunicationCheck(mcClient.DiscoveryClient)
+	if err != nil {
+		return nil, fmt.Errorf("communication with K8s API server failed.: %w", err)
```

Comment:
suggest: We might want to `return nil, err` instead of adding redundant info.

Reply:
👍  will do